breadth_first_search/BreadthFirstSearch.java

```diff
@@ -0,0 +1,65 @@
+import java.util.*;
+
+public class BreadthFirstSearch <T> {
+
+    // HasMap  of lists for Adjacency List Representation
+    public HashMap<Integer, ArrayList<T>> adj;
```

Comment:
When I put T-type then it gives error of compatible. Can you please check yours locally system?